File: cpp_1000-/2104.cc
// C++
// leetcode 2104
// https://leetcode-cn.com/problems/sum-of-subarray-ranges/
// 拙劣的双循环

class Solution {
public:
	using ll = long long;
    long long subArrayRanges(vector<int>& nums) {
        ll res = 0;
        int maxi, mini, n = nums.size();
        for(int i = 0; i < n; i++) {\
        	maxi = nums[i], mini = nums[i];
        	for(int j = i + 1; j < n; j++) {
        		maxi = max(maxi, nums[j]);
        		mini = min(mini, nums[j]);
        		res += maxi - mini;
        	}
        }
        return res;
    }
};

File: Go/Go_1000-/2104.go
// Go
// leetcode 2104
// https://leetcode.cn/problems/sum-of-subarray-ranges/

func subArrayRanges(nums []int) (res int64) {
	ma, mi, n := 0, 0, len(nums)
	for i, x := range nums {
		ma, mi = x, x
		for j := i + 1; j < n; j++ { // 暴力
			ma = max(ma, nums[j])
			mi = min(mi, nums[j])
			res += int64(ma - mi)
		}
	}
	return res
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}